test/e2e/testutils/tester/tester.go

```diff
@@ -16,105 +16,143 @@ package tester
 
 import (
 	"context"
+	"fmt"
 	"os"
+	"path/filepath"
+	"strconv"
 
+	capsulev1alpha1 "github.com/clastix/capsule/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	"github.com/liqotech/liqo/pkg/utils"
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	netv1alpha1 "github.com/liqotech/liqo/apis/net/v1alpha1"
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	sharingv1alpha1 "github.com/liqotech/liqo/apis/sharing/v1alpha1"
+	virtualKubeletv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+	testutils "github.com/liqotech/liqo/test/e2e/testutils/util"
 )
 
 // Tester is used to encapsulate the context where the test is executed.
 type Tester struct {
 	Clusters  []ClusterContext
 	Namespace string
+	// ClusterNumber represents the number of available clusters
+	ClusterNumber int
+	// the key is the clusterID and the value is the corresponding client
+	ClustersClients map[string]client.Client
```

Comment:
Isn't this redundant with respect to the Clusters array (which could be converted to map if necessary)?